Init commit

smrbanker opened the pull request:
MainActivity - главный класс проекта, его визуализация activity_dz. В нем закомментированы вызов Toast, двумя способами - через анонимный класс и лямбду
MediaActivity - класс для перехода с главного экрана, его визуализация activity_media SearchActivity - класс для перехода с главного экрана, его визуализация activity_search SettingsActivity - класс для перехода с главного экрана, его визуализация activity_settings, где реализован возврат на главный экран.

activity_main - UI для первого д/з (картинка из властелина колец)

Остальные файлы - это тренировка в процессе обучения. Повторял на практике различные задачи.

smrbanker committed:
Init commit

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,30 @@
+class SettingsActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
```
Обязательно:
Включая `EdgeToEdge` режим, также необходимо адаптировать экран под него. В противном случае, зона `Toolbar` заезжает под `ActionBar` из-за чего экран перестаёт соответствовать макетам и не возможно нажать на кнопку "Назад".
В частности, добавить следующий кусок кода, который будет проставлять отступы для `ActionBar`, чтобы элементы приложения не пересекались с ним.

```
ViewCompat.setOnApplyWindowInsetsListener(yourView) { view, insets ->
    val statusBar = insets.getInsets(WindowInsetsCompat.Type.statusBars())
    view.updatePadding(top = statusBar.top)
    insets
}
```

**Аналогично** для остальных экранов

IliaAnK commented on app/src/main/res/drawable/icon_media_.xml:
```diff
@@ -0,0 +1,12 @@
+    android:viewportHeight="26">
+  <path
+      android:pathData="M96.5,3H84.5C83.4,3 82.5,3.9 82.5,5V17C82.5,18.1 83.4,19 84.5,19H96.5C97.6,19 98.5,18.1 98.5,17V5C98.5,3.9 97.6,3 96.5,3ZM94.5,8H91.5V13.5C91.5,14.88 90.38,16 89,16C87.62,16 86.5,14.88 86.5,13.5C86.5,12.12 87.62,11 89,11C89.57,11 90.08,11.19 90.5,11.51V6H94.5V8ZM80.5,7H78.5V21C78.5,22.1 79.4,23 80.5,23H94.5V21H80.5V7Z"
+      android:fillColor="#1A1B22"/>
```
Рекомендация:
Цвета в векторных картинках тоже можно выносить в ресурсы, и писать `@color/<цвет>`

IliaAnK commented on app/src/main/res/drawable/poster.png:
Рекомендация:
Не нужный для этого проекта ресурс, можно удалить

IliaAnK commented on app/src/main/res/layout/activity_dz.xml:
```diff
@@ -0,0 +1,83 @@
+    android:background="@color/blue"
+    android:orientation="vertical">
+
+    <FrameLayout
```
Рекомендация:
В качестве заголовка экрана лучше использовать `MaterialToolbar`. Это стандартный компонент. В него можно передать текст заголовка с помощью атрибута `title`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.

IliaAnK commented on app/src/main/res/layout/activity_dz.xml:
```diff
@@ -0,0 +1,83 @@
+            />
+    </FrameLayout>
+
+    <LinearLayout
```
Рекомендация:
На мой взгляд этот `LinearLayout` лишний, попробуй переверстать экран только с одним корневым `LinearLayout`.

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,168 @@
+    android:background="@color/white"
+    android:orientation="vertical">
+
+   <LinearLayout
```
Рекомендация:
В качестве заголовка экрана лучше использовать `MaterialToolbar`. Это стандартный компонент. В него можно передать текст заголовка с помощью атрибута `title`, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль и атрибута `titleTextAppearance`.

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,168 @@
+       </FrameLayout>
+    </LinearLayout>
+
+    <LinearLayout
```
Рекомендация:
На мой взгляд этот `LinearLayout` лишний, попробуй переверстать экран только с одним корневым `LinearLayout`.

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,168 @@
+                android:paddingTop="@dimen/text_M"
+                android:paddingBottom="@dimen/text_M"
+                android:paddingLeft="@dimen/text_S"
+                android:text="Темная тема"
```
Обязательно:
Все строки, которые видит пользователь должны находится в ресурсах

IliaAnK commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,168 @@
+        android:layout_marginTop="24dp"
+        android:orientation="vertical">
+
+        <FrameLayout
```
Рекомендация к каждой конструкции
```
<FrameLayout> 
    <TextView/>
    <Switch или Button/> 
</FrameLayout>
```
Во-первых, это излишняя вложенность, которая негативно сказывается при отрисовке фрейма.
Во-вторых, она вызывает трудности с понимание того как свёрстан экран;
В-третьих, проще формировать и применять стили.

Каждая такая конструкция может быть представлена одной View.
Для свичера:
```
<com.google.android.material.switchmaterial.SwitchMaterial
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:gravity="center_vertical"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/darkTheme"
    android:textSize="16sp" />
```
Для TextView с иконкой справа:
```
<com.google.android.material.textview.MaterialTextView
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:drawableEnd="@drawable/icon_share_"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/share"
    android:textSize="16sp" />
```

Итого иерархия всего `layout`-а будет вот такой:
```
<LinearLayout>
    <MaterialToolbar/>
    <SwitchMaterial/>
    <MaterialTextView/>
    <MaterialTextView/>
    <MaterialTextView/>
<LinearLayout/>
```

IliaAnK commented on app/src/main/res/layout/activity_dz.xml:
```diff
@@ -0,0 +1,83 @@
+            android:layout_weight="1"
+            android:backgroundTint="@color/white"
+            app:cornerRadius="@dimen/med_but"
+            app:icon="@drawable/icon_search_"
```
Рекомендация:
Большинство `View` поддерживают атрибут `text`, именно через него и нужно задавать его
Для иконки можно использовать атрибут `drawableStart`

IliaAnK commented on app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,30 @@
+        val image4 = findViewById<Button>(R.id.button_back)
+
+        image4.setOnClickListener {
+            val displayIntent = Intent(this, MainActivity::class.java)
```
Обязательно:
Вместо создания и запуска новой активити:
```
val displayIntent = Intent(this, MainActivity::class.java)
startActivity(displayIntent)
```
нужно использовать метод `finish()` для закрытия текущей

smrbanker committed:
Спасибо, исправил все критические замечания. Все понятно!

Касательно рекомендаций, использовал один вариант в файле "activity_settings" на примере MaterialTextView. На досуге докручу все остальные элементы. Мы пока их не проходили, так что требует доп внимания.

IliaAnK commented:
Для стилизации свичера нужно:
1. Создать файл `res/color/switch_thumb_color.xml` для настройки цвета “ползунка” (thumb) в зависимости от состояния.
Внутри будет примерно следующее содержание:
```
<?xml version="1.0" encoding="utf-8"?>
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <!-- Активное состояние (включен) -->
    <item android:color="@color/switch_thumb_active_color" android:state_checked="true" />
    <!-- Неактивное состояние (выключен) -->
    <item android:color="@color/switch_thumb_inactive_color" android:state_checked="false" />
</selector>
```

2. Создать файл `res/color/switch_track_color.xml` для настройки цвета фона переключателя (track) в зависимости от состояния.
Внутри будет примерно следующее содержание:
```
<?xml version="1.0" encoding="utf-8"?>
<selector xmlns:android="http://schemas.android.com/apk/res/android">
    <!-- Активное состояние (включен) -->
    <item android:color="@color/switch_thumb_active_color" android:state_checked="true" />
    <!-- Неактивное состояние (выключен) -->
    <item android:color="@color/switch_thumb_inactive_color" android:state_checked="false" />
</selector>
```

3. И в XML-макете activity_settings для `SwitchMaterial` и примените созданные стили с помощь атрибутов `thumbTint` и :
```
app:thumbTint="@color/switch_thumb_color"
app:trackTint="@color/switch_track_color"
```
